src/cider/nrepl/middleware/reload.clj

```diff
@@ -0,0 +1,49 @@
+(ns cider.nrepl.middleware.reload
+  (:require
+   [clojure.main :refer [repl-caught]]
+   [clojure.string :as str]
+   [haystack.analyzer :as stacktrace.analyzer]
+   [nrepl.middleware.interruptible-eval :refer [*msg*]]
+   [nrepl.middleware.print :as print]
+   [nrepl.misc :refer [response-for]]
+   [nrepl.transport :as transport]
+   [orchard.misc :as misc]))
+
+(defn- user-reload
+  "clj-reload.core/reload from the user project.
+  Must be configured via clj-reload.core/init before being called."
+  []
+  (some-> (symbol "clj-reload.core" "reload")
+          resolve))
+
+(defn respond
+  [{:keys [transport] :as msg} response]
+  (transport/send transport (response-for msg response)))
+
+(defn- refresh-reply
+  [{:keys [::print/print-fn transport session id] :as msg}]
+  (let [{:keys [exec]} (meta session)]
+    (exec id
+          #(try
+             (let [reload (user-reload)]
+               (when-not reload
+                 (throw (ex-info "clj-reload.core/reload not found" {})))
+               (reload (cond-> {:log-fn (fn [& args]
+                                          (respond msg {:progress (str/join " " args)}))}
+                               (:all msg) (assoc :only :all)))
+               (respond msg {:status :ok}))
+             (catch Throwable error
+               (respond msg {:status :error
+                             :error  (stacktrace.analyzer/analyze error print-fn)})
```

Comment:
```suggestion
                             ;; TODO assoc :file, :line info if available
                             :error  (stacktrace.analyzer/analyze error print-fn)})
```

I'd like this to eventually return file/line info if we can easily grab it from the analyzer. This way cider.el could automatically open the culprit file.

(Feel free to give it a shot, but simply adding the TODO is perfectly fine)

Reply:
Added the TODO for now, but may circle back later if I can manage it.